Detectors/ZDC/reconstruction/src/DigiReco.cxx

```diff
@@ -0,0 +1,785 @@
+// Copyright CERN and copyright holders of ALICE O2. This software is
+// distributed under the terms of the GNU General Public License v3 (GPL
+// Version 3), copied verbatim in the file "COPYING".
+//
+// See http://alice-o2.web.cern.ch/license for full licensing information.
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+#include <TMath.h>
+#include "FairLogger.h"
+#include "ZDCReconstruction/DigiReco.h"
+#include "ZDCReconstruction/RecoParamZDC.h"
+
+namespace o2
+{
+namespace zdc
+{
+using O2_ZDC_DIGIRECO_FLT = float;
+
+void DigiReco::init()
+{
+  // Load configuration parameters
+  auto& sopt = ZDCSimParam::Instance();
+  mIsContinuous = sopt.continuous;
+  mNBCAHead = mIsContinuous ? sopt.nBCAheadCont : sopt.nBCAheadTrig;
+
+  if (!mModuleConfig) {
+    LOG(FATAL) << "Missing ModuleConfig configuration object";
+    return;
+  }
+
+  // Prepare tapered sinc function
+  // tsc/TSN =3.75 (~ 4) and TSL*TSN*sqrt(2)/tsc >> 1 (n. of sigma)
+  const O2_ZDC_DIGIRECO_FLT tsc = 750;
+  int n = TSL * TSN;
+  for (int tsi = 0; tsi <= n; tsi++) {
+    O2_ZDC_DIGIRECO_FLT arg1 = TMath::Pi() * O2_ZDC_DIGIRECO_FLT(tsi) / O2_ZDC_DIGIRECO_FLT(TSN);
+    O2_ZDC_DIGIRECO_FLT fs = 1;
+    if (arg1 != 0)
+      fs = TMath::Sin(arg1) / arg1;
+    O2_ZDC_DIGIRECO_FLT arg2 = O2_ZDC_DIGIRECO_FLT(tsi) / tsc;
+    O2_ZDC_DIGIRECO_FLT fg = TMath::Exp(-arg2 * arg2);
+    mTS[n + tsi] = fs * fg;
+    mTS[n - tsi] = mTS[n + tsi]; // Function is even
+  }
+
+  if (mTreeDbg) {
+    // Open debug file
+    mDbg = new TFile("ZDCReco.root", "recreate");
+    mTDbg = new TTree("zdcr", "ZDCReco");
```

Comment:
You better use `unique_ptr`, see e.g.
https://github.com/AliceO2Group/AliceO2/blob/7a6c7f384bddc76e787b6efa32c5a0842c661cf1/Detectors/Vertexing/include/DetectorsVertexing/PVertexer.h#L155-L156
https://github.com/AliceO2Group/AliceO2/blob/7a6c7f384bddc76e787b6efa32c5a0842c661cf1/Detectors/Vertexing/src/PVertexer.cxx#L615-L616
https://github.com/AliceO2Group/AliceO2/blob/7a6c7f384bddc76e787b6efa32c5a0842c661cf1/Detectors/Vertexing/src/PVertexer.cxx#L638-L644